test/rmath.jl

```diff
@@ -17,7 +17,18 @@ end
 get_statsfun(fname) = eval(Symbol(fname))
 get_rmathfun(fname) = eval(Meta.parse(string("RFunctions.", fname)))
 
-function rmathcomp(basename, params, X::AbstractArray, rtol=100eps(float(one(eltype(X)))))
+function rmathcomp(basename, params, X::AbstractArray, rtol=nothing)
+    # compute desired tolerance:
+    # has to take into account `params` as well since otherwise e.g. `X::Array{<:Rational}`
+    # always uses a tolerance based on `eps(one(Float64))` even when parameters are of type
+    # Float32
+    rtol = if rtol === nothing
+        PT = mapreduce(typeof, promote_type, params)
+        100 * eps(float(one(promote_type(PT, eltype(X)))))
+    else
+        rtol
+    end
```

Comment:
```suggestion
    if rtol === nothing
        PT = mapreduce(typeof, promote_type, params)
        rtol = 100 * eps(float(one(promote_type(PT, eltype(X)))))
    end
```

Reply:
I prefer the version in the PR since in my opinon it makes it easier to reason about the value of `rtol` and is easier to read.